lib/data/external/AwsClient.js

```diff
@@ -356,6 +358,41 @@ class AwsClient {
             return callback();
         });
     }
+    copyObject(request, sourceAwsBucketName, sourceObjectKey, log, callback) {
+        // for backwards compatibility
+        const destBucketName = request.bucketName;
+        const destObjectKey = request.objectKey;
+        const awsKey = this._createAwsKey(destBucketName, destObjectKey,
+          this._bucketMatch);
+        const metadataDirective = request.headers['x-amz-metadata-directive'];
+        const metaHeaders = trimXMetaPrefix(getMetaHeaders(request.headers));
+        this._client.copyObject({
+            Bucket: this._awsBucketName,
+            Key: awsKey,
+            CopySource: `${sourceAwsBucketName}/${sourceObjectKey}`,
+            Metadata: metaHeaders,
+            MetadataDirective: metadataDirective,
+        }, err => {
+            if (err) {
+                if (err.code === 'AccessDenied') {
+                    logHelper(log, 'error', 'Unable to access ' +
+                    `${sourceAwsBucketName} AWS bucket`, err,
```

Comment:
how do you know the access denied is due to the source rather than the destination?

Reply:
because the destination's AWS account copies the source.